src/expandable-section/expandable-section-header.tsx

```diff
@@ -242,11 +244,8 @@ export const ExpandableSectionHeader = ({
     );
   }
 
-  if (headerDescription && variant !== 'container' && variant !== 'default' && isDevelopment) {
-    warnOnce(
-      componentName,
-      'The `headerDescription` prop is only supported for the "default" and "container" variants.'
-    );
+  if (headerDescription && !variantSupportsDescription(variant) && isDevelopment) {
+    warnOnce(componentName, 'The `headerDescription` prop is not supported for this variant.');
```

Comment:
Can we output the variant name ? 
The headerDescription prop is not supported for ${variant} variant.

Reply:
👍 Done.